development/gardener-rotate/main.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/option"
+	secretmanager "google.golang.org/api/secretmanager/v1"
+	authentication "k8s.io/api/authentication/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	log = logrus.New()
+)
+
+type ServiceAccount struct {
+	KubernetesSA        string `json:"serviceAccount"`
+	KubernetesNamespace string `json:"namespace,omitempty"`
+	GCPSecret           string `json:"secret"`
+	GCPProject          string `json:"project"`
+	KeepOld             bool   `json:"keepOld,omitempty"`
+	Duration            int64  `json:"duration"`
+}
+
+type ConfigFile struct {
+	ServiceAccounts []ServiceAccount `json:"serviceAccounts"`
+}
+
+// Config stores command line arguments
+type Config struct {
+	ServiceAccount string
+	Kubeconfig     string
+	ConfigFile     string
+	Cluster        string
+	DryRun         bool
+}
+
+func main() {
```

Comment:
Please make code for rotating GCP secret a library. This part can be used then to rotate another secrets. Make approach standardised and avoid code duplication.